editoast/src/models/auth_driver.rs

```diff
@@ -100,14 +114,31 @@ impl StorageDriver for PgAuthDriver {
 
     #[tracing::instrument(skip_all, fields(%user_id), ret(level = Level::DEBUG), err)]
     async fn get_user_info(&self, user_id: i64) -> Result<Option<UserInfo>, Self::Error> {
+        #[derive(QueryableByName)]
+        struct UserIdentityFull {
+            #[diesel(sql_type = diesel::sql_types::Varchar)]
+            name: String,
+            #[diesel(sql_type = diesel::sql_types::Array<diesel::sql_types::Nullable<diesel::sql_types::Varchar>>)]
+            identities: Vec<Option<String>>,
+        }
         let conn = self.pool.get().await?;
-        let info = authn_user::table
-            .select((authn_user::identity_id, authn_user::name))
-            .filter(authn_user::id.eq(user_id))
-            .first::<(String, String)>(conn.write().await.deref_mut())
+        let raw_query = r"
+                SELECT u.name, ARRAY_AGG(i.identity) AS identities
+                FROM authn_user AS u
+                LEFT JOIN authn_user_identity AS i
+                ON u.id = i.user_id
+                WHERE u.id = $1
+                GROUP BY u.id, u.name;
+            ";
```

Comment:
Did you try to use diesel instead of a raw query. It's possible to run `group by` queries. Check doc [here](https://docs.diesel.rs/2.0.x/diesel/associations/trait.GroupedBy.html).

Reply:
I believe diesel chocked on `ARRAY_AGG`.